lib/apis/archive-files.js

```diff
@@ -93,13 +93,28 @@ module.exports = class ArchiveFilesAPI {
     }
     if (aborted) return
 
+    // read the manifest (it's needed in a couple places)
+    var manifest
+    try { manifest = await pda.readManifest(archive) } catch (e) { manifest = null }
+
     // find an entry
     var filepath = decodeURIComponent(reqPath)
     if (!filepath) filepath = '/'
     var isFolder = filepath.endsWith('/')
     var entry
     const tryStat = async (path) => {
       if (entry) return
+      // apply the web_root config
+      if (manifest && manifest.web_root) {
+        if (path) {
+        //   path = joinPaths(manifest.web_root, path)
```

Comment:
Can you use `joinPath` here? You can import it this way: `const joinPath = require('path').join`